packages/snap/src/keyring.test.ts

```diff
@@ -297,6 +475,38 @@ describe('Keyring', () => {
         expect(op).toStrictEqual(expected);
         expect(op.result.initCode).toBe(initCode);
       });
+
+      it('should throw an error for a number of transactions that is not 1', async () => {
+        const intents = [
+          [
+            {
+              to: '0x97a0924bf222499cBa5C29eA746E82F230730293',
+              value: '0x00',
+              data: ethers.ZeroHash,
+            },
+            {
+              to: '0x97a0924bf222499cBa5C29eA746E82F230730293',
+              value: '0x00',
+              data: ethers.ZeroHash,
+            },
+          ],
+          [],
+        ];
+
+        for (const intent of intents) {
+          await expect(
+            keyring.submitRequest({
+              id: 'ef70fc30-93a8-4bb0-b8c7-9d3e7732372b',
+              scope: '',
```

Comment:
Lets start adding  the CAIP2 to the scope and then having the keyring check if it supports the chain.

Reply:
Good idea! Addressed here [b1c79b4](https://github.com/MetaMask/snap-account-abstraction-keyring/pull/51/commits/b1c79b49bf1efa55601dd70d3bc4dba153b226d9)